tests/qnodes/test_qnode_base.py

```diff
@@ -521,11 +522,11 @@ def test_bad_wire_argument(self, operable_mock_device_2_wires):
         """Error: wire arguments must be intergers."""
 
         def circuit(x):
-            qml.RX(x, wires=[0.5])
+            qml.RX(x, wires=[qml.PauliX])
```

Comment:
😆 

(Out of curiosity, what was the reason for changing it from 0.5? Shouldn't that also raise an error?)

Reply:
Nope. At the moment the user can choose pretty much any number or string as a wire. We can change this if it's too loose. But if you think about it, we should not care what they call their wires, as long as we can identify the index at which this object sits in the device's record of all wires. PennyLane will only really work with the indices.

Reply:
Maybe we should constrain Wires from Numbers to Integral type?

Reply:
I wonder if there are odd edge cases... will the class treat `1.` and `1` as two separate wires?

Reply:
YOu are right. I will restrict the objects that can represent wires to Integrals in a separate PR!